src/pages/studyView/StudyViewPage.tsx

```diff
@@ -104,15 +105,24 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
         reaction(
             () => props.routing.location.query,
             query => {
+                let newStudyIds:string[]|null = null;
                 if ('studyId' in query) {
-                    this.store.studyIds = (query.studyId as string).split(",");
+                    newStudyIds = (query.studyId as string).split(",");
                 }
                 if ('id' in query) {
-                    this.store.studyIds = (query.id as string).split(",");
+                    newStudyIds = (query.id as string).split(",");
                 }
-                if (this.store.studyIds) {
-                    this.store.sampleAttrIds = ('sampleAttrIds' in query ? (query.sampleAttrIds as string).split(",") : []);
-                    this.store.patientAttrIds = ('patientAttrIds' in query ? (query.patientAttrIds as string).split(",") : []);
+                if (newStudyIds !== null) {
+                    const newStudyIdsMap = stringListToSet(newStudyIds);
+                    const existingStudyIdsMap = stringListToSet(this.store.studyIds);
+                    if (!_.isEqual(newStudyIdsMap, existingStudyIdsMap)) {
+                        // if new study ids in routing location, update store
+                        runInAction(()=>{
+                            this.store.studyIds = newStudyIds!;
+                            this.store.sampleAttrIds = ('sampleAttrIds' in query ? (query.sampleAttrIds as string).split(",") : []);
```

Comment:
could we change sampleAttrIds independent of studyIds?  If so, this wouldn't work right?   Also, we should perform same deep equality check on these and only update if unequal

Reply:
I got rid of `sampleAttrIds` and `patientAttrIds` as we are not using this feature in production.